src/reconcile/rec_row.c

```diff
@@ -295,7 +295,8 @@ __rec_row_merge(
         /* Update compression state. */
         __rec_key_state_update(r, false);
 
-        if (*build_delta && ref_changes > 0) {
+        /* Don't build page deltas for overflow keys. */
+        if (*build_delta && ref_changes > 0 && key->buf.size <= S2BT(session)->maxleafkey) {
```

Comment:
I think the change is wrong. We should not build overflow keys for full page as well.